### PR TITLE
solidity: update to 0.8.24

### PR DESCRIPTION
**Pull request by kaimingguo**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.7.3 21H1015 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: lang/solidity/Portfile [hunk 1]
```diff
@@ -24,36 +23,20 @@ long_description    Solidity is a statically typed, contract-oriented, \
...
-set port_libfmt     libfmt9
-
-depends_build-append \
-                    port:${port_libfmt} \
-                    port:range-v3
-depends_lib         port:jsoncpp
-
-cmake.module_path-append \
-                    ${prefix}/lib/${port_libfmt}/cmake
+depends_lib-append  port:boost
```
Comment (reneeotten) [inline lang/solidity/Portfile:31]: please keep using the "boost" PortGroup; what's the reason you changed?

File: lang/solidity/Portfile [hunk 2]
```diff
@@ -4,10 +4,9 @@ PortSystem          1.0
 
 PortGroup           github 1.0
 PortGroup           cmake 1.1
-PortGroup           boost 1.0
 
-github.setup        ethereum solidity 0.8.19 v
-github.tarball_from downloads
+github.setup        ethereum solidity 0.8.24 v
+github.tarball_from releases
 distname            ${github.project}_${github.version}
```
Comment (reneeotten) [inline lang/solidity/Portfile:10]: do you need to set the "distname" - typically the "github" PortGroup takes care of these things.

File: lang/solidity/Portfile [hunk 3]
```diff
@@ -24,36 +23,20 @@ long_description    Solidity is a statically typed, contract-oriented, \
...
                     2017
 
 configure.args      -DTESTS:BOOL=OFF \
                     -DBoost_USE_STATIC_LIBS:BOOL=OFF \
-                    -DWITH_Z3:BOOL=OFF \
-                    -DWITH_CVC4:BOOL=OFF \
-                    -DUSE_LD_GOLD:BOOL=OFF \
-                    -DRANGE_V3_INCLUDE_DIR:PATH="${prefix}/include" \
-                    -DJSONCPP_INCLUDE_DIR:PATH="${prefix}/include/jsoncpp" \
-                    -DJSONCPP_LIB_DIR:PATH="${prefix}/lib"
```
Comment (reneeotten) [inline lang/solidity/Portfile:53]: ideally we don't want to use vendored packages but the ones that MacPorts installs. Is the port not using any of the dependencies listed here anymore?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: lang/solidity/Portfile (hunk 1 above)
Comment (kaimingguo) [reply lang/solidity/Portfile:31]: When I used PortGroup to compile, Boost could not be linked correctly.  So I used `depends_lib` to link boost from MacPorts.

File: lang/solidity/Portfile (hunk 2 above)
Comment (kaimingguo) [reply lang/solidity/Portfile:10]: The default is `${github.project}-${github.version}.

ref: https://guide.macports.org/#reference.portgroup.github

File: lang/solidity/Portfile (hunk 3 above)
Comment (kaimingguo) [reply lang/solidity/Portfile:53]: The latest version of the required dependencies is managed through CMake, which requires a special version and may not be suitable for use with MacPorts.

Comment (reneeotten) [reply lang/solidity/Portfile:53]: I'm not sure what you mean. It either needs the dependencies or it doesn't. So presumably after this change it is using vendored versions these packages, that is not what we want. Specifying with CMake variables and point to ${prefix} should make sure the MacPorts versions are used. 

Please check this again as well as the boost dependency, that should ideally work with the PortGroup.

**Comment by reneeotten** [comment]
@kaimingguo ping

**Comment by kaimingguo** [comment]
Sorry, I'm busy with work at the moment. I need to test `boost` with PortGroup again.

**Comment by reneeotten** [comment]
sure - feel free to reopen this PR once you had the chance to work on it.

**Pull request closed without merging**